api/src/main/java/org/openmrs/module/appointments/dao/impl/AppointmentDaoImpl.java

```diff
@@ -117,4 +118,13 @@ public Appointment getAppointmentByUuid(String uuid) {
         }
         return criteria.list();
     }
+
+    @Override
+    public List<Appointment> search(AppointmentSearch appointmentSearch) {
```

Comment:
The function seems to suggest that it takes a generic AppointmentSearch object, which I am assuming is created for future extensibility purpose. However, subsequently in the code, we seem to assume that the start and end date are always given. 
- Are these start and end dates mandatory for search? If not, then we should add criteria conditionally.